orca-kayenta/src/main/kotlin/com/netflix/spinnaker/orca/kayenta/tasks/RunKayentaCanaryTask.kt

```diff
@@ -37,16 +40,45 @@ class RunKayentaCanaryTask(
 
   override fun execute(stage: Stage): TaskResult {
     val context = stage.mapTo<RunCanaryContext>()
+    // The `DeployCanaryClusters` stage will deploy a list of experiment/control
+    // pairs, but we will only canary the first pair in `deployedClusters`.
+    val scopes = stage.context["deployedClusters"]?.let {
+      val clusters = OrcaObjectMapper.newInstance().convertValue<List<DeployedCluster>>(it)
+      context.scopes.from(clusters.first())
+    } ?: context.scopes
+
     val canaryPipelineExecutionId = kayentaService.create(
       context.canaryConfigId,
       stage.execution.application,
       stage.execution.id,
       context.metricsAccountName,
       context.storageAccountName /* configurationAccountName */, // TODO(duftler): Propagate configurationAccountName properly.
       context.storageAccountName,
-      CanaryExecutionRequest(context.scopes, context.scoreThresholds)
+      CanaryExecutionRequest(scopes, context.scoreThresholds)
     )["canaryExecutionId"] as String
 
     return TaskResult(SUCCEEDED, singletonMap("canaryPipelineExecutionId", canaryPipelineExecutionId))
   }
 }
+
+private fun Map<String, CanaryScopes>.from(cluster: DeployedCluster): Map<String, CanaryScopes> {
+  return entries.associate { (key, scope) ->
+    key to scope.copy(
+      controlScope = scope.controlScope.copy(
+        scope = cluster.controlScope,
+        location = cluster.controlLocation
+      ),
+      experimentScope = scope.experimentScope.copy(
+        scope = cluster.experimentScope,
+        location = cluster.experimentLocation
+      )
+    )
+  }
+}
+
+internal data class DeployedCluster(
```

Comment:
Does this really describe a cluster? Looks like a pair of server groups that are likely in the same cluster I suppose.